scripts/verify-rules/__init__.py

```diff
@@ -0,0 +1,120 @@
+import json
+import subprocess
+import glob
+import os
+import yaml
+
+
+class DummyYamlLoader(yaml.SafeLoader):
+    '''
+    Custom loader to handle custom tags (like !ImportValue) in yaml files.
+    They are ignored (their value is set to None) as they are not used by this script
+    '''
+
+    def __init__(self, stream) -> None:
+        super().__init__(stream)
+        self.add_constructor(
+            "!ImportValue", DummyYamlLoader.dummy_yaml_constructor)
+        self.add_constructor(
+            "!Sub", DummyYamlLoader.dummy_yaml_constructor)
+
+    def dummy_yaml_constructor(loader, node):
+        return None
+
+
+def run_cmd(cmd):
+    result = subprocess.run(cmd, capture_output=True)
+    return result.stdout.decode("utf-8")
+
+
+def get_aws_rules(startToken: str = None) -> "list[dict]":
+    '''
+    Get the list of rules in AWS. May recursively call itself to aggregate
+    a list to handle pagination.
+    '''
+
+    cmds = ["aws", "events", "list-rules", "--max-items", "30", "--starting-token",
+            startToken] if startToken else ["aws", "events", "list-rules", "--max-items", "30"]
+
+    raw = run_cmd(cmds)
+    jsonData = json.loads(raw)
+    rules = jsonData["Rules"]
+
+    if "NextToken" in jsonData:
+        nextToken = jsonData["NextToken"]
+        rules = rules + get_aws_rules(nextToken)
+
+    return rules
+
+
+def parse_local_rules(ymlPath: str) -> "list[str]":
+    '''
+    Parse a local serverless.yml file and extract its event-bridge rules.
+
+    Name follows a format like <service>-dev-<function>-rule-<index> in order
+    to approximate actual rule names.
+    '''
+
+    parsed = None
+    with open(ymlPath, mode="r", encoding="utf-8") as ymlFile:
+        parsed = yaml.load(ymlFile, DummyYamlLoader)
+
+    if parsed:
+        ymlDirName = os.path.split(os.path.split(ymlPath)[0])[1]
+        print(f"parsing {ymlPath}")
+
+        if "functions" in parsed:
+            functions = parsed["functions"]
+            return [f"{ymlDirName}-dev-{funcName}-rule-{i+1}" for funcName in functions.keys() if "events" in functions[funcName] for i in range(len([e for e in functions[funcName]["events"] if "eventBridge" in e]))]
+
+    print(f"unable to parse {ymlPath}")
+    return []
+
+
+def get_local_rules() -> "list[str]":
+    ymlFiles = glob.glob(
+        "../../services/**/serverless.yml")
+
+    print(f"found {len(ymlFiles)} local serverless.yml files")
+    return [rule for ymlPath in ymlFiles for rule in parse_local_rules(ymlPath)]
+
+
+# Helper script to verify discrepancies for event-bridge rules between remote (aws) and local definitions
+if __name__ == "__main__":
+    existing_rules = get_aws_rules()
```

Comment:
To be a bit more pythonic you could put all this in an main function and den in the if block call main()